WIP: General Fresh Start

eternal44 opened the pull request:
Name files to be useful for letting us use continuous testing, thinking
about using Guard.

Starting over, to help us think about the contract, and ease of
including partial library, think: Facets

kotp committed:
WIP: General Fresh Start

Name files to be useful for letting us use continuous testing, thinking
about using Guard.

Starting over, to help us think about the contract, and ease of
including partial library, think: Facets

eternal44 commented on lib/unit_conversion/distance_conversion.rb:
```diff
@@ -1,44 +0,0 @@
-module DistanceConversion
```
Is distance the right name for this file?  Should we use length, another unit name?

kotp commented:
The names need to be some kind of consistent for using a continuous testing process.  Guard is a decent choice for the purpose of testing.

eternal44 committed:
Remove auto-generated bin folder

If we need this in the future we'll make it again.  There was nothing
executable in it and the gem didn't depend on it either.

eternal44 committed:
Using retro-versioning for our rebuild.

Using 0.0.10 to signify major change.  Didn't have a cleaner way of
writing version numbers so...  this is our solution.

eternal44 committed:
WIP: Guardfile setup

Configuring Guardfile regex to match file structure

kotp committed:
WIP: Guard file pattern updated

Removed all tests... we can make a new test.

This is still FAIL

eternal44 committed:
Make Guardfile watch /lib folder

eternal44 committed:
Setup Minitest environment

Converting previous Spec environment to Minitest.

eternal44 committed:
Change Rakefile to run Minitest

Same as last commit: it was looking for spec files to run.

eternal44 committed:
WIP: Sanity check for Minitest.

Minitest runs properly but I'm having trouble pulling in the conversion
file to run the tests.  The problem seems to be my pathing.

eternal44 committed:
WIP: Guardfile setup.

Doesn't run when I save conversion file changes but runs fine whenever
I save a file change to my tests.

eternal44 committed:
Revert back to minitest::spec

Mistakenly converted our minitest::spec environment to a unit test one.
Changing it back.

eternal44 committed:
WIP Bugfix:  fix pathing to require distance in test

My distance formula is still wrong.

kotp committed:
Allows us to not rely on Guard to run a test

Guard with the guard minitest helps us have minitest in our testing
environment.

The spec helper provides that dependency for anyone that simply wants to
run a test, without relying on a tool they may not be familiar with, or
not in need of a continuous testing environment for that moment.

kotp committed:
Organizing namespace and flesh out API

We are still experiementing with what the tool should be.  And how to
use it.  This experimenting with a module and class combination.
